### ISSUE TITLE
.html Extension Removed from Tag File URLs

### ISSUE THREAD

**Issue by AntumDeluge** [doxygen/doxygen#9075]
I've been trying to figure out how to link to external documentation for hours. I finally learned how to format & use tag files. But the problem is that the `.html` extension is being trimmed from URLs when Doxygen creates the links.

Here is my tag file:
```xml
<?xml version='1.0' encoding='UTF-8' standalone='yes' ?>
<tagfile doxygen_version="1.9.3" doxygen_gitid="c0b9eafbfb53286ce31e75e2b6c976ee4d345473">

	<compound kind="class">
		<name>java::lang::Object</name>
		<filename>java.base/java/lang/Object.html</filename>
	</compound>
	<compound kind="class">
		<name>java::util::List</name>
		<filename>java.base/java/util/List.html</filename>
	</compound>
	<compound kind="class">
		<name>java::util::Set</name>
		<filename>java.base/java/util/Set.html</filename>
	</compound>

</tagfile>
```

And in my `Doxyfile` I have: `TAGFILES = ./buildtools/doxygen/tags/java_api.xml=https://docs.oracle.com/en/java/javase/16/docs/api/`

But instead of linking to https://docs.oracle.com/en/java/javase/16/docs/api/java.base/java/lang/Object.html it links to https://docs.oracle.com/en/java/javase/16/docs/api/java.base/java/lang/Object.

Doxygen version:
```
doxygen --version
1.9.3 (c0b9eafbfb53286ce31e75e2b6c976ee4d345473)
```

---
**Edit:** Downgrading to v1.8.12 solves the problem.

**Comment by albert-github** [comment]
Downgrading to 1.8.12 is of course not the solution, the problem should be found and fixed.
In your issue you already specify the used tag file and the used setting in the `Doxyfile`, still it is better to have a complete example:
- Can you please attach a, small, self contained example (source+configuration file in a tar or zip) that allows us to reproduce the problem? Please don't add external links as they might not be persistent.

**Comment by AntumDeluge** [comment]
@albert-github I believe this is what you asked for. The main page in the generated HTML docs links to a method parameter with the stripped URL referencing `java.lang.String`. Let me know if there is something else you need me to provide.

[doxygen_sample.zip](https://github.com/doxygen/doxygen/files/7940507/doxygen_sample.zip)

**Comment by albert-github** [comment]
Minimal is a bit different, but I see the error here in the places where `String` is referenced.

**Comment by albert-github** [comment]
You wrote in the initial issue:
> Downgrading to v1.8.12 solves the problem.

though with version 1.8.12 I don't see a link at all. The first time I see a link is in 1.9.2 and here it is indeed already wrong.

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #9076 

(a small, quick, workaround might be to set in your current project `HTML_FILE_EXTENSION` to `.xhtml`)

**Comment by AntumDeluge** [comment]
There may be more to this issue. I just created a new tags file for linking to API at http://luaj.org/luaj/3.0/api/. The `.html` extensions are not stripped from these links. Could it have something to do with linking to https://docs.oracle.com/ specifically?

I read issues with previous versions that an extra `.html` extension was being appended to some links. I am wondering if these included any links other than to docs.oracle.com.

**Comment by albert-github** [comment]
There have been a number of changes regarding the extensions and some iterations took place.

The problem in this issue had clearly to do with the fact hat doxygen discovered an extension in a link although there was none, so the patch is certainly something that is correct.
Based on your comment I revisited example and I see that when I change (in my small example) the `java.base` into `java_base` that the extensions remains (tested with the 1.9.3 version). It looks like the tag reader does something incorrect, I'll have to investigate (thanks for the idea).

Code of #9075 has been integrated in master on GitHub (please don't close the issue as we have to check the "tag reader problem").

**Comment by albert-github** [comment]
I checked the "tag reader" problem and the fact that the extension remains missing was a typical "https://docs.oracle.com/" special as here in the path there is a `.` in the directory name.
In all cases the extension is, correctly, stripped from the filename in case the extension is in the tag file the same as in the current run of doxygen.

**Conclusion**: the fix as made in the pull request #9075 is the correct fix and has been integrated in master on GitHub.
(please don't close the issue as this will be done at the moment of an official release)

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.4.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
